File: beer-bar.js
var eventBus = require("vertx/event_bus");

eventBus.registerHandler("bar", function(message, replier) {
  java.lang.System.err.println("BAR: Someone ordered " + message.amount + " beer(s)");
  replier({wait_time: message.amount * 1.75});
});

java.lang.System.err.println("The BAR is open!");

File: BeerBar.java
import org.vertx.java.core.Handler;
import org.vertx.java.core.eventbus.Message;
import org.vertx.java.core.json.JsonObject;
import org.vertx.java.platform.Verticle;

public class BeerBar extends Verticle {

  public void start() {
    vertx.eventBus().registerHandler("bar", new Handler<Message<JsonObject>>() {
      @Override
      public void handle(Message<JsonObject> message) {
        String name = message.body().getString("name");
        String strAmount = message.body().getString("amount");
        System.out.println("BAR: " + name  + " ordered " + strAmount + " beer(s)");

        double time = 1.75 * Integer.parseInt(strAmount);

        JsonObject json = new JsonObject();
        json.putNumber("wait_time", time);
        message.reply(json);
      }
    });
    System.out.println("The BAR is open!");
  }
}
